root/types/edit_types.js

```diff
@@ -0,0 +1,1095 @@
+/*
+ * @flow strict
+ * Copyright (C) 2017 MetaBrainz Foundation
+ *
+ * This file is part of MusicBrainz, the open internet music database,
+ * and is licensed under the GPL version 2, or (at your option) any
+ * later version: http://www.gnu.org/licenses/gpl-2.0.txt
+ */
+
+declare type AddAnnotationEditT = $ReadOnly<{
+  ...GenericEditT,
+  +display_data: {
+    +changelog: string,
+    +entity_type: AnnotatedEntityTypeT,
+    [annotatedEntityType: AnnotatedEntityTypeT]: AnnotatedEntityT,
```

Comment:
This line causes a flow error:

`Cannot create 'EditNote' element because in property 'edit': Either indexed  object type [1] is incompatible with exact  object type [2] in property 'display_data'. [etc]`

Is that solvable without splitting the types into `AddArtistAnnotationEditT`, `AddLabelAnnotationEditT`, etc?